tests/hmm/hmumu_utils.py

```diff
@@ -1188,6 +1204,38 @@ def get_bit_values(array, bit_index):
     """
     return (array & 2**(bit_index)) >> 1
 
+# Custom kernels to get the number of softJets with pT>5 GEV
+def nsoftjets(nevt,softjets, leading_muon, subleading_muon, leading_jet, subleading_jet, ptcut, dr2cut):
+    njet_out = NUMPY_LIB.zeros(nevt, dtype=NUMPY_LIB.int32)
+    nsoftjets_cpu(nevt, softjets.offsets, softjets.pt, softjets.eta, softjets.phi, leading_jet["eta"], subleading_jet["eta"], leading_jet["phi"], subleading_jet["phi"], leading_muon["eta"], subleading_muon["eta"], leading_muon["phi"], subleading_muon["phi"], ptcut, dr2cut, njet_out)
+    return njet_out
+
+@numba.njit(parallel=True, fastmath=True)
+def nsoftjets_cpu(nevt, softjets_offsets, pt, eta, phi, etaj1, etaj2, phij1, phij2, etam1, etam2, phim1, phim2, ptcut, dr2cut, njet_out):
```

Comment:
If you want, you could also achieve deltaR matching using the special function `mask_deltar_first`, here is an example: https://github.com/jpata/hepaccelerate-cms/blob/master/tests/hmm/hmumu_utils.py#L130.

Reply:
Yes, will try to use this function later.